### PR TITLE
✨Source Microsoft Teams Migrate Python CDK to Low-code CDK

### LINKED ISSUE #36198: ✨ Source Microsoft Teams: Migrate Python CDK to Low-code CDK
### Topic

Migrate Microsoft Teams Source to Low-code CDK

### Relevant information

https://docs.airbyte.com/integrations/sources/microsoft-teams

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source Microsoft Teams to Low-Code CDK*
Closes airbytehq/airbyte-internal-issues#6812

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## Tests

<details open><summary><strong>Integration & Acceptance</strong></summary>

 - Full passed state:
 
image

</details>

## 🚨 User Impact 🚨
- Spec Breaking changes, migration to low-code

### CODE REVIEW

**Comment by btkcodedev** [comment]
1) Built with connector builder :white_check_mark: 
2) Error handling used :white_check_mark: 
3) Built and Ran tests from `airbyte-ci` :white_check_mark: 

Full test results from `airbyte-ci`:
image

**Comment by btkcodedev** [comment]
Ready for review :white_check_mark:

File: airbyte-integrations/connectors/source-microsoft-teams/metadata.yaml [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
-    pypi:
-      enabled: true
-      packageName: airbyte-source-microsoft-teams
-  registries:
-    cloud:
-      enabled: true
-    oss:
-      enabled: true
+  name: Microsoft Teams
+  releaseDate: 2024-04-18
```
Comment (girarda) [inline airbyte-integrations/connectors/source-microsoft-teams/metadata.yaml:28]: let's set this to `2020-12-21`

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,5801 @@
...
+              - string
+        type:
+          - 'null'
+          - object
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://graph.microsoft.com/v1.0/
+        path: users?$top={{ config['page_size'] }}
```
Comment (girarda) [inline airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml:73]: let's set a default value
`path: users?$top={{ config.get('page_size', 20) }}`

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,5801 @@
...
+spec:
+  connection_specification:
+    $schema: http://json-schema.org/draft-07/schema#
+    type: object
+    required:
+      - client_id
+      - client_secret
+      - period
+      - tenant_id
+    properties:
```
Comment (girarda) [inline airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml:5747]: let's move these properties in a `credentials` block so it's compatible with the existing spec

Comment (btkcodedev) [reply airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml:5741]: Done, It's backward compatible now

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml (hunk 2 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml]: Done

File: airbyte-integrations/connectors/source-microsoft-teams/metadata.yaml (hunk 1 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-microsoft-teams/metadata.yaml]: Done!

**Comment by btkcodedev** [comment]
All comments resolved :white_check_mark: 
Results: All pass :white_check_mark: 
image

**Comment by btkcodedev** [comment]
Inline schema ✅
Poetry lock file ✅
Breaking change referred in metadata ✅
Add migration.md docs file ✅
Docs updated ✅

`airbyte-ci test` - All pass
image

**Comment by btkcodedev** [comment]
Marking PR as ready to merge

**Comment by btkcodedev** [comment]
Continued in https://github.com/airbytehq/airbyte/pull/36950

**Pull request closed without merging**